### ISSUE TITLE
Possible for relation type embedding?

### ISSUE THREAD

**Issue by Bin-Go2** [facebookresearch/PyTorch-BigGraph#165]
Hi, here is my puzzle, 

when I do converting process, it could create a relation_types_parameters.tsv file which contains the parameters of the operators for the relation types. What I am curious about is whether the parameters of the operators for a specific relation type can represent the embedding of the relation type.

**Comment by lw** [comment]
This is a rather open question, as the meaning of "embedding" is somewhat loosely defined. In a nutshell: I guess it depends on what you want to do with those embeddings afterwards, and ultimately you can only find out if it works for you by trial-and-error.

To be more precise: I am assuming when you mean "using relation params as embeddings" you mean something like comparing the parameters of two relations to determine how "similar" they are. (E.g., in a knowledge base, the "was born in" and "grew up in" relations will probably be highly correlated). In that case, the bad news is that PBG (and AFAIK systems like it) were not designed with such a usage in mind and thus do _not_ optimize for it. It's still possible that this can work, as a "side-effect", but in that case it would be accidental.

In order to do this properly, I guess you would first need to formally define what you want to capture. That is, come up with an evaluation metric so that you can compare several approaches and know what works best for you. This would however only give you an "after-the-fact" insight: the model itself would still train to optimize the similarity of entity embeddings. And changing that is much harder: you'd need to figure out a way to calculate a loss function that aligns with your intention and this could possibly require a different way of sampling negatives and several other assumptions might break too. You could risk having to overhaul the entire system.

**Comment by Bin-Go2** [comment]
Oh, got it. Thanks for your patient explanation! 😄
